include/crpropa/magneticField/turbulentField/SimpleGridTurbulence.h

```diff
@@ -0,0 +1,115 @@
+	                         double sIndex = 5. / 3)
+	    : TurbulenceSpectrum(Brms, lMin, lMax, 0, sIndex, 0) {}
+	~SimpleTurbulenceSpectrum() {}
+
+	/**
+	General energy spectrum for synthetic turbulence models
+	*/
+	double energySpectrum(double k) const {
+		return std::pow(k, -getSindex() - 2);
+	}
+
+	/**
+	    @brief       compute the magnetic field coherence length according to
+	   the formula in  Harari et Al JHEP03(2002)045
+	    @return Lc   coherence length of the magnetic field
+	*/
+	double getCorrelationLength() const {
+		return turbulentCorrelationLength(getLmin(), getLmax(),
+		                                  -getSindex() - 2);
+	}
+	static double turbulentCorrelationLength(double lMin, double lMax,
+	                                         double s) {
+		double r = lMin / lMax;
+		return lMax / 2 * (s - 1) / s * (1 - pow(r, s)) / (1 - pow(r, s - 1));
+	}
+};
+
+/**
+ @class SimpleGridTurbulence
+ @brief Turbulent grid-based magnetic field with a simple power-law spectrum
+ */
+class SimpleGridTurbulence : public GridTurbulence {
+  public:
+	/**
+	 Create a random initialization of a turbulent field.
+	 @param spectrum    TurbulenceSpectrum instance to define the spectrum of
+	 turbulence
+	 @param gridProp	GridProperties instance to define the underlying grid
+	 @param seed	 Random seed
+	 */
+	SimpleGridTurbulence(const SimpleTurbulenceSpectrum &spectrum,
+	                     const GridProperties &gridProp, unsigned int seed = 0);
+
+	static void initTurbulence(ref_ptr<Grid3f> grid, double Brms, double lMin,
+	                           double lMax, double alpha, int seed);
+};
+
+// Compatibility with old functions from GridTurbulence:
+
+/** Analytically calculate the correlation length of the simple model turbulent
+ * field */
+inline double turbulentCorrelationLength(double lMin, double lMax,
+                                         double alpha = -11 / 3.) {
+	KISS_LOG_WARNING
+	    << "turbulentCorrelationLength is deprecated and will be "
+	       "removed in the future. Replace it with an appropriate "
+	       "turbulent field model and call getCorrelationLength().";
+	return SimpleTurbulenceSpectrum::turbulentCorrelationLength(lMin, lMax,
+	                                                            -alpha - 2);
+}
+
+/**
+ Create a random initialization of a turbulent field.
+ @param lMin	Minimum wavelength of the turbulence
+ @param lMax	Maximum wavelength of the turbulence
+ @param alpha	Power law index of <B^2(k)> ~ k^alpha (alpha = -11/3 corresponds
+ to a Kolmogorov spectrum)
+ @param Brms	RMS field strength
+ @param seed	Random seed
+ */
+inline void initTurbulence(ref_ptr<Grid3f> grid, double Brms, double lMin,
+                           double lMax, double alpha = -11 / 3., int seed = 0) {
+	KISS_LOG_WARNING
+	    << "initTurbulence is deprecated and will be removed in the future. "
+	       "Replace it with an appropriate turbulent field model instance.";
```

Comment:
It will be replaced with a more appropriate...

Reply:
Changed.